Mage.Sets/src/mage/cards/m/MoreOfThatStrangeOil.java

```diff
@@ -0,0 +1,59 @@
+package mage.cards.m;
+
+import mage.MageObject;
+import mage.abilities.Mode;
+import mage.abilities.effects.common.CounterTargetEffect;
+import mage.abilities.effects.common.counter.ProliferateEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.TargetController;
+import mage.filter.FilterPermanent;
+import mage.filter.FilterSpell;
+import mage.filter.predicate.Predicate;
+import mage.filter.predicate.Predicates;
+import mage.target.TargetPermanent;
+import mage.target.TargetSpell;
+
+import java.util.UUID;
+
+/**
+ * @author tiera3 - based on GetOut
+ */
+public final class MoreOfThatStrangeOil extends CardImpl {
+
+    private static final FilterSpell filter = new FilterSpell("creature, artifact, or planeswalker spell");
+    private static final Predicate<MageObject> predicate = Predicates.or(
```

Comment:
no point in declaring this predicate as a field, just add it directly where it's used in the static block below